### DISCUSSION TITLE
Doubt on VVenC 1.5.0 not reaching informed target bitrate

### DISCUSSION THREAD

**Discussion by cecosme** [fraunhoferhhi/vvenc#184] [Q&A] [answered]
Dear all,

First: congratulation on the excellent job at VVenC/VVdeC.

I am playing with VVenC 1.5.0 for a hypothetical scenario where the target video bitrate is 10 Mbps for a CBR channel and measuring VMAF scores.

The first content I am using (Novela_1080p.yuv) has the following characteristics: 1920x1080p, 59.94 fps, SDR, BT.2020, 10-bit depth, and planar YUV.

The VVenC's Powershell command line I am using is:

.\vvencapp.exe -i Novela_1080p.yuv -s 1920x1080 --format yuv420_10 --framerate 60000 --framescale 1001 --preset faster 
--profile main_10 --level 4.1 --tier high --bitrate 10000000 --passes 1 --qpa 0 --gopsize 32 --refreshsec 1 
--threads 8 --hdr off -o Novela_1080p_10000000.266 | tee Novela_1080p_10000000_ENC.log

The header from "Novela_1080p_10000000_ENC.log" file is:

	vvencapp: Fraunhofer VVC Encoder ver. 1.5.0 [Windows][VS 1930][64 bit][SIMD=AVX2]
	CODING TOOL CFG: CTU64 QT44BTT00 IBD:0 SAO:1 ALF:1 (NonLinLuma:0 NonLinChr:0) CCALF:1 Tiles:1x1 Slices:1 WPP:0 WPP-Substreams:1 TMVP:1 DQ:0 SDH:1 CST:1 BDOF:0 DMVR:1 MTSImplicit:1 SBT:0 JCbCr:0 CabacInitPresent:0 AMVR:0 SMVD:0 LMCS:0 CIIP:0 MIP:0 AFFINE:0 MMVD:0 SbTMVP:0 GPM:0 LFNST:0 MTS:0 ISP:0 TS:2 TSLog2MaxSize:3 useChromaTS:0 BDPCM:2 IBC:2 BCW:0 
	ENC. ALG. CFG: QPA:0 HAD:1 (fast) RDQ:2 RDQTS:1 ASR:1 MinSearchWindow:96 RestrictMESampling:0 EDO:0 MCTF:2 
	PRE-ANALYSIS CFG: STA:1 LeadFrames:0 TrailFrames:0 
	FAST TOOL CFG: ECU:1 FEN:1 FDM:1 FastSearch:4 (SCC:2) LCTUFast:1 FastMrg:2 PBIntraFast:2 AMaxBT:0 FastQtBtEnc:1 ContentBasedFastQtbt:1 FastIntraTools:0 IntraEstDecBit:3 FastLocalDualTree:1 IntegerET:1 FastSubPel:1 QtbttExtraFast:7 FastTTSplit:0 IBCFastMethod:6 FIM:4 (29) ALFSpeed:1 QuantThr: 4 
	RATE CONTROL CFG: RateControl:1 Passes:1 Pass:-1 TargetBitrate:10000000 RCInitialQP:0 RCForceIntraQP:0 LookAhead:1 
	PARALLEL PROCESSING CFG: NumThreads:8 MaxParallelFrames:4 WppBitEqual:1 WF:0

The tail from "Novela_1080p_10000000_ENC.log" file is:

	Total Frames |   Bitrate     Y-PSNR    U-PSNR    V-PSNR    YUV-PSNR   
	     3598    a    5936.8923   50.7116   53.1514   53.6680   51.3040

The question is: why is the VVenC bitrate for Novela_1080p only reach 5936 kbps if the Target Bitrate from command line is 10000 kbps?

I have two other video contents with the same video characteristics as Novela_1080p.yuv (but they are more complex to encode) and using the same VVenC command line I can reach final bitrates closer to the Target Bitrate (like 10109 kbps and 10148 kbps).

Thanks in advance.

**Comment by adamjw24** [comment, marked as answer]
Thanks for reporting.

This is known bahavior that we are looking to fix in an upcoming release, probably towards the end of the month.

For now the workaround would be to use the 2-pass rate control (`--passes 2`).

**Reply by cecosme** [reply]
Dear Adam, thanks for the feedback.
